nano/core_test/node.cpp

```diff
@@ -4300,161 +4300,176 @@ TEST (rep_crawler, local)
 	node_flags.enable_pruning = true;
+
 	auto & node1 = *system.add_node (node_config, node_flags);
-	nano::keypair key1;
-	auto send1 = nano::send_block_builder ()
-				 .previous (nano::dev::genesis->hash ())
+	nano::keypair key1{};
+	nano::send_block_builder builder{};
+	auto latest_hash = nano::dev::genesis->hash ();
+
+	auto send1 = builder.make_block ()
+				 .previous (latest_hash)
 				 .destination (key1.pub)
 				 .balance (nano::dev::constants.genesis_amount - nano::Gxrb_ratio)
 				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
-				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .work (*system.work.generate (latest_hash))
 				 .build_shared ();
-	auto send2 = nano::send_block_builder ()
-				 .previous (send1->hash ())
+	node1.process_active (send1);
+
+	latest_hash = send1->hash ();
+	auto send2 = builder.make_block ()
+				 .previous (latest_hash)
 				 .destination (key1.pub)
 				 .balance (0)
 				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
-				 .work (*system.work.generate (send1->hash ()))
+				 .work (*system.work.generate (latest_hash))
 				 .build_shared ();
-	// Process as local blocks
-	node1.process_active (send1);
 	node1.process_active (send2);
-	node1.block_processor.flush ();
-	node1.scheduler.flush ();
-	// Confirm last block to prune previous
-	{
-		auto election = node1.active.election (send1->qualified_root ());
-		ASSERT_NE (nullptr, election);
-		election->force_confirm ();
-	}
-	ASSERT_TIMELY (2s, node1.block_confirmed (send1->hash ()) && node1.active.active (send2->qualified_root ()));
+
+	// Force-confirm both blocks
+	node1.process_confirmed (nano::election_status{ send1 });
+	ASSERT_TIMELY (5s, node1.block_confirmed (send1->hash ()));
+	node1.process_confirmed (nano::election_status{ send2 });
+	ASSERT_TIMELY (5s, node1.block_confirmed (send2->hash ()));
+
+	// Three blocks in total, nothing pruned yet
 	ASSERT_EQ (0, node1.ledger.cache.pruned_count);
-	{
-		auto election = node1.active.election (send2->qualified_root ());
-		ASSERT_NE (nullptr, election);
-		election->force_confirm ();
-	}
-	ASSERT_TIMELY (2s, node1.active.empty () && node1.block_confirmed (send2->hash ()));
+	ASSERT_EQ (3, node1.ledger.cache.block_count);
+
 	// Pruning with default age 1 day
 	node1.ledger_pruning (1, true, false);
 	ASSERT_EQ (0, node1.ledger.cache.pruned_count);
 	ASSERT_EQ (3, node1.ledger.cache.block_count);
+
 	// Pruning with max age 0
-	node1.config.max_pruning_age = std::chrono::seconds (0);
+	node1.config.max_pruning_age = std::chrono::seconds{ 0 };
 	node1.ledger_pruning (1, true, false);
 	ASSERT_EQ (1, node1.ledger.cache.pruned_count);
 	ASSERT_EQ (3, node1.ledger.cache.block_count);
+
 	ASSERT_TRUE (node1.ledger.block_or_pruned_exists (nano::dev::genesis->hash ()));
-	ASSERT_TRUE (node1.ledger.block_or_pruned_exists (send1->hash ())); // true for pruned
+	ASSERT_TRUE (node1.ledger.block_or_pruned_exists (send1->hash ()));
```

Comment:
I wouldn't remove this comment as it is there to facilitate debugging.

Reply:
I did add some relevant comments at the beginning of the tests.